url6f34cc586e_mekku93_sxl_interpreter_repl_tgz-pJ8-com_miguel_sxl_SXLJ8/src/com/miguel/sxl/SXLParserTokenManager.java

```diff
@@ -0,0 +1,931 @@
+         if ((active0 & 0x8000000000000L) != 0L)
+         {
+            jjmatchedKind = 55;
+            jjmatchedPos = 5;
+            return 21;
+         }
+         return -1;
+      case 6:
+         if ((active0 & 0x8000000000000L) != 0L)
+         {
+            jjmatchedKind = 55;
+            jjmatchedPos = 6;
+            return 21;
+         }
+         return -1;
+      default :
+         return -1;
+   }
+}
+private final int jjStartNfa_0(int pos, long active0)
+{
+   return jjMoveNfa_0(jjStopStringLiteralDfa_0(pos, active0), pos + 1);
+}
+private int jjStopAtPos(int pos, int kind)
+{
+   jjmatchedKind = kind;
+   jjmatchedPos = pos;
+   return pos + 1;
+}
+private int jjMoveStringLiteralDfa0_0()
+{
+   switch(curChar)
+   {
+      case 33:
+         return jjMoveStringLiteralDfa1_0(0x400000L);
+      case 35:
+         return jjStopAtPos(0, 32);
+      case 40:
+         return jjStopAtPos(0, 26);
+      case 41:
+         return jjStopAtPos(0, 27);
+      case 42:
+         return jjStopAtPos(0, 15);
+      case 43:
+         return jjStopAtPos(0, 13);
+      case 44:
+         return jjStopAtPos(0, 25);
+      case 45:
+         return jjStopAtPos(0, 14);
+      case 47:
+         return jjStartNfaWithStates_0(0, 16, 23);
+      case 58:
+         return jjStopAtPos(0, 23);
+      case 59:
+         return jjStopAtPos(0, 24);
+      case 60:
+         jjmatchedKind = 18;
+         return jjMoveStringLiteralDfa1_0(0x400000000100000L);
+      case 61:
+         jjmatchedKind = 12;
+         return jjMoveStringLiteralDfa1_0(0x200000L);
+      case 62:
+         jjmatchedKind = 17;
+         return jjMoveStringLiteralDfa1_0(0x80000L);
+      case 97:
+         return jjMoveStringLiteralDfa1_0(0x200000000000L);
+      case 98:
+         return jjMoveStringLiteralDfa1_0(0x4000000000L);
+      case 99:
+         return jjMoveStringLiteralDfa1_0(0x8000000000L);
+      case 101:
+         return jjMoveStringLiteralDfa1_0(0x2000000000000L);
+      case 102:
+         return jjMoveStringLiteralDfa1_0(0x8000000000000L);
+      case 104:
+         return jjMoveStringLiteralDfa1_0(0x40000000000000L);
+      case 105:
+         return jjMoveStringLiteralDfa1_0(0x1801000000000L);
+      case 108:
+         return jjMoveStringLiteralDfa1_0(0x80000000000L);
+      case 110:
+         return jjMoveStringLiteralDfa1_0(0x400000000000L);
+      case 111:
+         return jjMoveStringLiteralDfa1_0(0x100000000000L);
+      case 114:
+         return jjMoveStringLiteralDfa1_0(0x10002000000000L);
+      case 115:
+         return jjMoveStringLiteralDfa1_0(0x50000000000L);
+      case 117:
+         return jjMoveStringLiteralDfa1_0(0x20000000000L);
+      case 119:
+         return jjMoveStringLiteralDfa1_0(0x24000000000000L);
+      case 123:
+         return jjStopAtPos(0, 28);
+      case 125:
+         return jjStopAtPos(0, 29);
+      default :
+         return jjMoveNfa_0(0, 0);
+   }
+}
+private int jjMoveStringLiteralDfa1_0(long active0)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 50 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.